test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperDisplayTextTest.cs

```diff
@@ -296,6 +297,34 @@ public void DisplayTextFor_IgnoresModelStateEntry()
             Assert.Equal("Property value", result);
         }
 
+        [Fact]
+        public void DisplayTextFor_EnumDisplayAttribute_WhenPresent()
+        {
+            // Arrange
+            var model = EnumWithDisplayAttribute.Value1;
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(model);
+
+            // Act
+            var result = helper.DisplayTextFor(m => m);
```

Comment:
The fix isn't specific to `DisplayTextFor(...)`. Two options:
1. Mix it up between these tests e.g. use `helper.DisplayText(expression: string.empty)` or `helper.DisplayText(expression: null)` in this test, leaving the other one alone
2. Get three results in `// Act` for the different helper calls and assert them all in `// Assert` for both tests.

Would also like to see a confirmation the fix isn't specific to identity expression or `null` / empty property name. Please add one test with an `EnumWithDisplayAttribute` property within a model.